### PR TITLE
Refactoring of MODNetModel

### PR DESCRIPTION
**Pull request by ml-evs**
This PR tidies up the `modnet.models` module:

- [x] Lots of cosmetic changes, e.g. docstrings, type hints and refactoring of methods
- [x] Removed some remaining defunct files from git-lfs
- [x] No longer refit with `fit_preset`, just cache the best model
- [x] Added tests for simple model training/fit_preset/save/load
   - Seems like there is a bug with `.save()` and Python 3.8, where some multiprocessing object cannot be pickled
   - Fixed annoying tensorflow forking problem by running tests with only 1 multiprocessing job
- [x] Added ability to pass a `MODData` as `val_data` in `fit()` which gets transformed into `validation_data` for Keras.
- [x] Fixed bug with MODData splitting and `get_` methods

### CODE REVIEW

File: modnet/models.py [hunk 1]
```diff
@@ -1,213 +1,452 @@
...
-        elif self.xscale == 'standard':
-            x = self.scaler.transform(x)
-        
+        logging.info(
+            "Preset #{} resulted in lowest validation loss.\nFitting all data...".format(
+                best_preset + 1
+            )
+        )
+        self.n_feat = best_n_feat
+        self.model = best_model
```
Comment (ppdebreuck) [inline modnet/models.py:349]: The only issue with not refitting, is that we do not use all the training data. Usually before testing, a refit is done on all data except test set.

**Review by ppdebreuck — CHANGES_REQUESTED** [review]
I would add a a refit on all the data (val_frac=0)

File: modnet/models.py (hunk 1 above)
Comment (ml-evs) [reply modnet/models.py:349]: It currently does the same thing though right? i.e. the previous fit call I removed also used `val_fraction=0.2` so it was just training on the same data again?

If I understand you correctly, you're suggesting to just tweak that call to use `val_fraction=0` yeah?

Comment (ppdebreuck) [reply modnet/models.py:349]: ow I see, that was a mistake, should have been val_frac=0 on the refit!

Comment (ppdebreuck) [reply modnet/models.py:349]: As discussed, probably best to leave at is (give some guarantee on the model + user can always refit if needed) but use val_fraction as argument with 0.1 default.

**Comment by ml-evs** [comment]
So, Python is segfaulting when it runs the preprocessing tests, depending on whether the model tests have also run... (model tests individually do not crash either). Trying to debug...

**Review by ppdebreuck — APPROVED** [review]
